src/client/Microsoft.Identity.Client/ClientApplicationBase.cs

```diff
@@ -85,6 +87,15 @@ public async Task<IEnumerable<IAccount>> GetAccountsAsync()
         {
             RequestContext requestContext = CreateRequestContext(Guid.NewGuid());
             IEnumerable<IAccount> accounts = Enumerable.Empty<IAccount>();
+
+#if ANDROID
```

Comment:
Please remove if #Android (see comment below). And unit test this piece of code - you need to assert that `broker.GetAccountsAsync(clientID)` is called. There already exist infrastructure to inject a mocked broker, so this should be fairly easy.

Reply:
yes but the PlatformProxy.BrokerSupportsSilentCall requires a mocked platform proxy. how do I achieve that?

Reply:
The way I would do it is to add an extra param to the existing `SetBrokerForTest` [hook](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/86c5572232f2657eba6eb1919a749874877a996b/src/client/Microsoft.Identity.Client/PlatformsCommon/Interfaces/IPlatformProxy.cs#L111)

`void SetBrokerForTest(IBroker broker, bool? supportsSilentFlow);`

It's important to use a nullable boolean because there are 3 states (null for not set, true or false for the possible values). Product uses null. Test uses values.